docs/program-setup/bare_metal/mbed2_porting.md

```diff
@@ -0,0 +1,125 @@
+# Porting an Mbed OS 2 target to Mbed OS 6 bare metal
+
+The Mbed OS bare metal profile removes the RTOS and provides fewer features compared to Mbed OS 6. This profile offers the same functionality as Mbed OS 2 and allows targets to access features that we have added to more recent versions of Mbed OS. This document describes how to configure an Mbed OS 2 target for bare metal and validate the port.
+
+## Configuring your target
+
+You will use the blinky bare metal Mbed OS example and make it run on your target. The configuration will be validated upon successful run of the example.
+
+The main steps are:
+1. Clone [mbed-os-example-blinky-baremetal](https://github.com/ARMmbed/mbed-os-example-blinky-baremetal).
+1. Edit `targets.json`. In this step, you will configure your target to support Mbed OS 6 and add bare metal configuration parameters.
+1. Build and run mbed-os-example-blinky-baremetal. In this step, you will validate the configuration and troubleshoot any issue.
+
+### Clone mbed-os-example-blinky-baremetal
+
+Run the following command:
+- `git clone https://github.com/ARMmbed/mbed-os-example-blinky-baremetal`
+
+And then change directory:
+- `cd mbed-os-example-blinky-baremetal`
+
+### Edit targets.json
+
+Find your target in `mbed-os/targets/targets.json` and update its configuration as described below.
+
+Configure your target to support Mbed OS 6:
+- Remove the `release_versions` property as it is no longer required.
+```
+"release_versions": ["2"]
+```
+
+- Add the `supported_application_profiles` property to indicate that the application profile supported by this target is bare metal.
+```
+"supported_application_profiles" : ["bare-metal"]
+```
+
+- Override `supported_c_libs` property to link with the smaller C libraries. The default for all targets is defined as follows: 
+```
+"supported_c_libs": {
+    "arm":  ["std"],
+    "gcc_arm":  ["std"], ["small"],
+    "iar": ["std"]
+}
+```
+
+Both the ARM and GCC_ARM toolchains support optimized versions of the C standard libraries, microlib and newlib-nano respectively. We recommend using them with the bare metal profile to achieve lower memory footprints. Ultra constrained targets should override `supported_c_libs` as follows:
+```
+"supported_c_libs": {
+    "arm":  ["small"],
+    "gcc_arm":  ["small"]
```

Comment:
How about `"iar"`? Were Mbed 2 targets not supported by the IAR toolchain?

Reply:
If IAR standard library is supported then it will be added in the next step.